lib/engines/hckinstall/hckinstall.rb

```diff
@@ -276,33 +274,50 @@ def run_second(client:)
     end
 
     def prepare_setup_scripts_config
-      kit_string = @platform['kit']
-      kit_type = kit_string[0..2]
-      kit_version = ''
-      kit_type == 'HCK' || kit_version = kit_string[3..]
+      kit_type, kit_version = parse_kit_info
 
       config = {
         kit_type:,
         hlk_kit_ver: kit_version
       }
 
-      unless @kit_info['download_url'].nil?
-        download_kit_installer(@kit_info['download_url'],
-                               "#{kit_type}#{kit_version}", @hck_setup_scripts_path)
+      @kit_path = find_kit(kit_type, kit_version)
+
+      if @kit_info['download_url'].nil? && @kit_path.nil?
+        raise(AutoHCKError, 'HLK installer download URL is not provided and installer is not found')
       end
```

Comment:
Why don't you raise in the following `if @kit_path.nil?`? This saves checking `@kit_path.nil?` twice.